backend/api/quivr_api/modules/sync/utils/syncutils.py

```diff
@@ -85,7 +84,17 @@ async def _upload_files(
                 mime_type = file.mime_type
                 modified_time = file.last_modified
 
-                file_response = self.sync_cloud.download_file(credentials, file)
+                if self.sync_cloud.lower_name == "notion":
```

Comment:
Here we have a change of parallelising the download of the files.
We can spawn multiple tasks and await on their return 
```suggestion
tasks = []
file_response = asyncio.gather(*[download_file( credentials, file) for file in files])

async def download_file():
    ...

```

Reply:
I think that the `self.sync_cloud.lower_name` isn't a robust for discriminating the correct download function. The issue here is having two methods for dealing with download : one sync and one async. Maybe have the syncs that don't have async method to be async of to just raise if we access their async method